src/io/flutter/view/InspectorPerfTab.java

```diff
@@ -76,8 +74,18 @@
 
     final FlutterWidgetPerfManager widgetPerfManager = FlutterWidgetPerfManager.getInstance(app.getProject());
 
-    showPerfOverlay = new BoolServiceExtensionCheckbox(app, SHOW_PERFORMANCE_OVERLAY, "Show performance overlay", "");
-    showRepaintRainbow = new BoolServiceExtensionCheckbox(app, SHOW_REPAINT_RAINBOW, "Show repaint rainbow", "");
+    showPerfOverlay =
+      new BoolServiceExtensionCheckbox(
+        app,
+        ServiceExtensions.performanceOverlay.getExtension(),
+        ServiceExtensions.performanceOverlay.getDisabledText(),
+        "");
+    showRepaintRainbow =
+      new BoolServiceExtensionCheckbox(
```

Comment:
make
BoolServiceExtensionCheckbox just take the whole extension description as an argument.
That way it can be later cleaned up to use more of the metadata provided and it will simplify this code a bit.